plenum/server/ledgers_bootstrap.py

```diff
@@ -0,0 +1,231 @@
+from typing import Any, List, Optional, NamedTuple, Dict
+
+from common.exceptions import LogicError
+from common.serializers.serialization import state_roots_serializer
+from crypto.bls.bls_bft import BlsBft
+from ledger.compact_merkle_tree import CompactMerkleTree
+from ledger.genesis_txn.genesis_txn_initiator_from_file import GenesisTxnInitiatorFromFile
+from plenum.common.constants import AUDIT_LEDGER_ID, POOL_LEDGER_ID, CONFIG_LEDGER_ID, DOMAIN_LEDGER_ID, \
+    NODE_PRIMARY_STORAGE_SUFFIX, BLS_LABEL, HS_MEMORY
+from plenum.common.ledger import Ledger
+from plenum.persistence.storage import initStorage
+from plenum.server.batch_handlers.audit_batch_handler import AuditBatchHandler
+from plenum.server.batch_handlers.config_batch_handler import ConfigBatchHandler
+from plenum.server.batch_handlers.domain_batch_handler import DomainBatchHandler
+from plenum.server.batch_handlers.pool_batch_handler import PoolBatchHandler
+from plenum.server.request_handlers.audit_handler import AuditTxnHandler
+from plenum.server.request_handlers.get_txn_author_agreement_aml_handler import GetTxnAuthorAgreementAmlHandler
+from plenum.server.request_handlers.get_txn_author_agreement_handler import GetTxnAuthorAgreementHandler
+from plenum.server.request_handlers.node_handler import NodeHandler
+from plenum.server.request_handlers.nym_handler import NymHandler
+from plenum.server.request_handlers.txn_author_agreement_aml_handler import TxnAuthorAgreementAmlHandler
+from plenum.server.request_handlers.txn_author_agreement_handler import TxnAuthorAgreementHandler
+from plenum.server.request_managers.action_request_manager import ActionRequestManager
+from plenum.server.request_managers.read_request_manager import ReadRequestManager
+from plenum.server.request_managers.write_request_manager import WriteRequestManager
+from state.pruning_state import PruningState
+
+from storage.helper import initHashStore, initKeyValueStorage
+from stp_core.common.log import getlogger
+
+logger = getlogger()
+
+IN_MEMORY_LOCATION = '//!memory!'
+
+# TODO: This can be improved in several ways
+#  - if register_batch_handler gets some improvements it will be possible to group
+#    functions by ledger instead of by initialization stage
+#  - create request managers and expose as properties instead of passing as parameters,
+#    just like bls bft
+#  - if we introduce some entity (named Executor, Application, BusinessLogic or whatever)
```

Comment:
I think that `WriteRequestManager` plays the role of executor now. It already contains `DatabseManager`.
As for `ReadRequestManager` and `ActionRequestManager`, I think they don't belong to `Executor` interface. This is separate entities which are independent of both consus and transaction execution logic (they can and should go into a separate service/process).

Reply:
Agree with point on moving them into separate processes. Just a couple of comments on that:
- if they are in different processes then they will need to open databases separately. Updates across different databases are not atomic, so some form of synchronization might be needed.
- even though they are in different processes they access same databases (and there are quite a lot of them), so it is still desirable to have common code at least for db initialization
- thinking about observers (running on a separate node) - while they serve only read requests they will most likely require WriteReqManager as well in order to build all those cache databases, and probably state as well